keyboards/omnikeyish/info.json

```diff
@@ -0,0 +1,12 @@
+{
```

Comment:
This looks good, I loaded it into configurator and got this layout
![Screen Shot 2019-06-20 at 13 05 53](https://user-images.githubusercontent.com/2275667/59878056-31020100-935c-11e9-8d43-f44bcee9cd3a.png)

Reply:
THe only issue, is that some of the keys are not included here, like between Esc and F1, so it will have issues, I believe.

Reply:
Used a KLE converter to convert it to a qmk conf file. The trouble is... the PCB supports quite a bit of button positions that this one doesn't show. With and without left Fkey block. ANSI enter, Big Arse Enter, ISO enter, ISO or ANSI left shifts, 1.5-1-1.5u-7u-1.5u-1u-1.5 or 1.25-1.25-1.25-6.25-1.25-1.25-1.25-1.25u bottom rows. Don't know how to represent that.

Reply:
@drashna there's no physical pads on the pcb for matrix key 1,4 (the spot between esc and f1. so it'd be impossible to generate that press.

Reply:
If you want to get crazy, you can look at the s7_elephant for hints.

Reply:
So... the default LAYOUT macro allows for some mutually exclusive choices. How do you recommend expressing that in the info.json? 

As an example. A Big Arse Enter key choice, makes it physically impossible fit a switch for normal ANSI 1.5u backslash key, or the ISO 'non-US backslash' key (in QMK speak), since the BA enter takes up the space where they're normally be. Likewise an ANSI enter makes the 'non-US backslash' imposible to fit, and an ISO Enter block the ANSI backslash location. The same thing is the case with left and right shift keys, and the bottom row modifier keys.

Reply:
Each layout macro in `omnikeyish.h` should have a corresponding JSON key in `info.json`, under the `layouts` tree.

The layout macro that contains every matrix position supported by the PCB - even if it's impossible to physically build the keyboard in that layout - should be named `LAYOUT_all`, and then the other layouts that *can* be physically built should define a subset of the matrix positions that `LAYOUT_all` defines.

The [Clueboard 66% rev4](https://github.com/qmk/qmk_firmware/blob/master/keyboards/clueboard/66/rev4/) is a good example to follow. Its header file [defines every matrix location on the PCB](https://github.com/qmk/qmk_firmware/blob/master/keyboards/clueboard/66/rev4/rev4.h#L32-L49), which [looks like this in QMK Configurator](https://user-images.githubusercontent.com/18669334/59939517-0fad1d80-940d-11e9-90eb-8478b811339c.png). The board can't be built as shown, but if you switch to `LAYOUT_66_ansi` or `LAYOUT_66_iso`, you get two keys where there were three before. Those layout macros delete either `k74` or `k65` as appropriate.

https://github.com/qmk/qmk_firmware/blob/7f23ad72c5c736b58bab16995e7b0a599268b56f/keyboards/clueboard/66/rev4/rev4.h#L34-L35

vs.

https://github.com/qmk/qmk_firmware/blob/7f23ad72c5c736b58bab16995e7b0a599268b56f/keyboards/clueboard/66/rev4/rev4.h#L55-L56

vs.

https://github.com/qmk/qmk_firmware/blob/7f23ad72c5c736b58bab16995e7b0a599268b56f/keyboards/clueboard/66/rev4/rev4.h#L76-L77

Reply:
Can the configurator handle a info.json entry with overlaping keys for LAYOUT_all?

Reply:
Not well. It'll render all the keys, but depending on ordering, you might have one key fully on top of another. I'd suggest going the Clueboard 66%'s route: ANSI Backslash at 1.5u, ISO Non-US Hash at 1u, and Enter at 1.25u.

Reply:
Welp... tried designing a info.json layout entry for each LAYOUT macro in KLE and converting them afterward. The ALL design has a not to scale pathetic 1.25u enter key, but shouldn't have any overlapping. Configurator people happy now?